outlook/mail/pkg/graph/mail.go

```diff
@@ -37,13 +38,24 @@ func GetMessageDetails(ctx context.Context, client *msgraphsdkgo.GraphServiceCli
 	return result, nil
 }
 
-func SearchMessages(ctx context.Context, client *msgraphsdkgo.GraphServiceClient, subject, fromAddress, fromName, folderID string) ([]models.Messageable, error) {
+func SearchMessages(ctx context.Context, client *msgraphsdkgo.GraphServiceClient, subject, fromAddress, fromName, folderID, start, end string) ([]models.Messageable, error) {
 	var (
 		result models.MessageCollectionResponseable
 		err    error
 		filter []string
 	)
 
+	// It is important that start and end filters are first in the list.
+	// Details in the first answer on this question:
+	// https://learn.microsoft.com/en-us/answers/questions/656200/graph-api-to-filter-results-on-from-and-subject-an
+	if end != "" {
```

Comment:
So you trust the LLM that end > start? :grin:

Reply:
Haha, so far it has not messed it up.